### PR TITLE
Kafka + gevent post

### PR DESCRIPTION
**Pull request by tsx**

### CODE REVIEW

**Review by AlecRosenbaum — APPROVED** [review]
Nicely written

File: _posts/2020-09-22-kafka-gevent.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Solutions
+
+What do we do about it? We went through a couple options:
+
+- Switch to a non-official pure-Python library. This carries additional risks like potential bugs that are not discovered because it’s used less and compatibility issues when it’s time to upgrade.
```
Comment (jkemp101) [inline _posts/2020-09-22-kafka-gevent.md:34]: Not sure if you want to change up this language slightly to not offend the other library. Just not sure I would consider the one we use official other than it does use the core C libraries under the hood. But it is produced by Confluent not Apache.

File: _posts/2020-09-22-kafka-gevent.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+![Profiling results](/assets/kafka-gevent/flush-profiler.png)
+
+## Threading
+
+In most migrations we had before, scaling out was not necessary, we could just run it in background for a while with minimal impact. Not this time though. The volume of data that needed processing was so large that running it in one thread would take multiple months, so we need to scale out. Performance metrics I've collected during initial runs indicated that impact will be minimal even if we run 10 migration processes. So in next iteration the migration process was changed to run in multiple parallel threads in the same process. At Close, the main threading framework we use in Python is `gevent`. It's a green threading library that hooks into IO routines and schedules other green threads - greenlets - inside the same OS thread to run whenever currently running thread needs to wait on IO operation to be complete.
```
Comment (jkemp101) [inline _posts/2020-09-22-kafka-gevent.md:18]: so we **needed** to scale out
So in **the** next iteration 
wait on **an** IO operation to be

**Review by jkemp101 — APPROVED** [review]
Added a few nit comments but looks great.

File: _posts/2020-09-22-kafka-gevent.md (hunk 1 above)
Comment (tsx) [reply _posts/2020-09-22-kafka-gevent.md]: Rewritten from the perspective of switching to a _different_ library and the need to for additional testing to match our requirements. This way it feels more neutral and doesn't imply it's worse in any way, just that we're not willing to take the risk of a larger change to a critical component without extra testing and consideration.

**Pull request merged**